connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/LiteralValueProtoConverter.scala

```diff
@@ -97,6 +101,56 @@ object LiteralValueProtoConverter {
     }
   }
 
+  def toLiteralProtoBuilder(
+      literal: Any,
+      dataType: DataType): proto.Expression.Literal.Builder = {
+    val builder = proto.Expression.Literal.newBuilder()
+
+    def arrayBuilder(list: List[_], elementType: DataType) = {
+      val ab = builder.getArrayBuilder.setElementType(toConnectProtoType(elementType))
+      list.foreach(x => ab.addElements(toLiteralProto(x, elementType)))
+      ab
+    }
+
+    def mapBuilder(map: Map[_, _], keyType: DataType, valueType: DataType) = {
+      val mb = builder.getMapBuilder
+        .setKeyType(toConnectProtoType(keyType))
+        .setValueType(toConnectProtoType(valueType))
+      map.foreach { case (k, v) =>
+        mb.addKeys(toLiteralProto(k, keyType))
+        mb.addValues(toLiteralProto(v, valueType))
+      }
+      mb
+    }
+
+    def structBuilder(values: Seq[Any], structType: StructType) = {
+      val sb = builder.getStructBuilder.setStructType(toConnectProtoType(structType))
+      values.zip(structType.fields.map(_.dataType)).foreach { case (v, dataType) =>
+        sb.addElements(toLiteralProto(v, dataType))
+      }
+      sb
+    }
+
+    (literal, dataType) match {
+      case (v: List[_], ArrayType(elementType, _)) =>
```

Comment:
for support both `mutable` and `immutable` collection, I think we should match `scala.collection.Seq` and `scala.collection.Map`